ggml-qnn.cpp

```diff
@@ -0,0 +1,3588 @@
+
+#define QNN_VER_PTR(x)                                  (&((x).v1))
+#define QNN_OP_CFG_VALID(op_config)                      ((op_config).version == QNN_OPCONFIG_VERSION_1)
+
+#define QNN_OP_CFG_GET_NAME(op_config)                   get_qnn_oponfig_name(op_config)
+#define QNN_OP_CFG_GET_PACKAGE_NAME(op_config)           get_qnn_op_config_packagename(op_config)
+#define QNN_OP_CFG_GET_TYPE_NAME(op_config)              get_qnn_op_config_typename(op_config)
+#define QNN_OP_CFG_GET_NUM_PARAMS(op_config)             get_qnn_op_config_numparams(op_config)
+#define QNN_OP_CFG_GET_PARAMS(op_config)                 get_qnn_op_config_params(op_config)
+#define QNN_OP_CFG_GET_NUM_INPUTS(op_config)             get_qnn_op_config_numinputs(op_config)
+#define QNN_OP_CFG_GET_INPUTS(op_config)                 get_qnn_op_config_inputs(op_config)
+#define QNN_OP_CFG_GET_NUM_OUTPUTS(op_config)            get_qnn_op_config_numoutputs(op_config)
+#define QNN_OP_CFG_GET_OUTPUTS(op_config)                get_qnn_op_config_outputs(op_config)
+
+#define QNN_OP_CFG_SET_NAME(op_config, value)            set_qnn_op_config_name(op_config, value)
+#define QNN_OP_CFG_SET_PACKAGE_NAME(op_config, value)    set_qnn_op_config_packagename(op_config, value)
+#define QNN_OP_CFG_SET_TYPE_NAME(op_config, value)       set_qnn_op_config_typename(op_config, value)
+
+#define QNN_OP_CFG_SET_PARAMS(op_config, num_of_params, params) \
+  set_qnn_op_config_params(op_config, num_of_params, params)
+
+#define QNN_OP_CFG_SET_INPUTS(op_config, num_of_inputs, inputTensors) \
+  set_qnn_op_config_inputs(op_config, num_of_inputs, inputTensors)
+
+#define QNN_OP_CFG_SET_OUTPUTS(op_config, num_of_outputs, output_tensors) \
+  set_qnn_op_config_outputs(op_config, num_of_outputs, output_tensors)
+
+#define QNN_TENSOR_GET_ID(tensor)                       get_qnn_tensorid(tensor)
+#define QNN_TENSOR_GET_NAME(tensor)                     get_qnn_tensorname(tensor)
+#define QNN_TENSOR_GET_TYPE(tensor)                     get_qnn_tensortype(tensor)
+#define QNN_TENSOR_GET_DATA_FORMAT(tensor)              get_qnn_tensor_dataformat(tensor)
+#define QNN_TENSOR_GET_DATA_TYPE(tensor)                get_qnn_tensor_datatype(tensor)
+#define QNN_TENSOR_GET_QUANT_PARAMS(tensor)             get_qnn_tensor_quantparams(tensor)
+#define QNN_TENSOR_GET_RANK(tensor)                     get_qnn_tensor_rank(tensor)
+#define QNN_TENSOR_GET_DIMENSIONS(tensor)               get_qnn_tensor_dimensions(tensor)
+#define QNN_TENSOR_GET_MEM_TYPE(tensor)                 get_qnn_tensor_memtype(tensor)
+#define QNN_TENSOR_GET_CLIENT_BUF(tensor)               get_qnn_tensor_clientbuf(tensor)
+#define QNN_TENSOR_GET_MEM_HANDLE(tensor)               get_qnn_tensor_memhandle(tensor)
+
+#define QNN_TENSOR_SET_ID(tensor, value)                set_qnn_tensor_id(tensor, value)
+#define QNN_TENSOR_SET_NAME(tensor, value)              set_qnn_tensor_name(tensor, value)
```

Comment:
Just curious whats the purpose of defining those macros, looks its a simple wrapper of the corresponding function here, but why don't we use those function directly?

Reply:
these macros are referenced from QNN Samples or https://github.com/pytorch/executorch/tree/main/backends/qualcomm which are both provided by Qualcomm Technologies, Inc.

I keep them here for further usage.

Reply:
Thank you for the reply.
Cannot found any similar symbol liked `QNN_TENSOR_SET_NAME` in the link your provided, but found it in another repo [QnnTypeMacros.hpp#L443](https://github.com/XiaoMi/StableDiffusionOnDevice/blob/65f09ae58a7c4a4c14aaeaaeaf5f5376a72f8b74/app/src/main/cpp/include/QnnTypeMacros.hpp#L443), looks they have a dediacted header `QnnTypeMacros.hpp`, don't know if its in the sdk, but from my personal aspect, its better to not put them in this file for better maintainability.

Reply:
Xiaomi's project is totally provided by QTI.

Reply:
yeah, what i mean is that if we don't use them currently, we should remove them here, and add them back when we actually need them.

Reply:
you are right if you are a full time employee in a commercial IT company.

as I said before, I keep them here for further potential usage.

Reply:
thought its not about who or what they do, its just some basic rule to make the PR clean and more easy to review, 
looks this glue code take 800+ lines, and as you said those're useless currently, IMO an unnecessary change.

Reply:
ok, I'll remove them accordingly.

Reply:
thanks!

Reply:
you are welcome.